Service/Menu/Cloner.php

```diff
@@ -103,8 +103,12 @@ public function clone(MenuInterface $menu): MenuInterface
 
             foreach ($this->menuNodes->getList($menu) as $node) {
                 $nodeClone = $this->nodeFactory->create();
+                $data = $node->getData();
 
-                $nodeClone->setData($node->getData());
+                if (isset($data['customer_groups'])) {
+                    $data['customer_groups'] = json_encode((array)$data['customer_groups']);
```

Comment:
In code where `customerGroups` is serialized, unserialized we are using `Magento\Framework\Serialize\SerializerInterface`

Let's keep this uniform across this module